ext/ffi/fast_call.rs

```diff
@@ -0,0 +1,1352 @@
+          ),
+          Signed(DW) | Unsigned(DW) => dynasm!(self.assembler
+            ; .arch aarch64
+            ;  ldr w8, [sp, self.offset_trampoline + padding_trampl]
+            ; str w8, [sp, self.offset_original + padding_orig]
+          ),
+          Signed(QW) | Unsigned(QW) => dynasm!(self.assembler
+            ; .arch aarch64
+            ; ldr x8, [sp, self.offset_trampoline + padding_trampl]
+            ; str x8, [sp, self.offset_original + padding_orig]
+          ),
+        }
+        self.offset_trampoline += padding_trampl + size_trampl;
+        self.offset_original += padding_orig + size_orig;
+      }
+    };
+  }
+
+  fn zero_first_arg(&mut self) {
+    dynasm!(self.assembler
+      ; .arch aarch64
+      ; mov x0, xzr
+    );
+  }
+
+  fn tailcall(&mut self, ptr: *const c_void) {
+    // stack pointer is never modified and remains aligned
+    // frame pointer remains the one provided by the trampoline's caller (V8)
+
+    let mut address = ptr as u64;
+    let mut imm16 = address & 0xFFFF;
+    dynasm!(self.assembler
+      ; .arch aarch64
+      ; movz x8, imm16 as u32
+    );
+    address >>= 16;
+    let mut shift = 16;
+    while address > 0 {
+      imm16 = address & 0xFFFF;
+      dynasm!(self.assembler
+        ; .arch aarch64
+        ; movk x8, imm16 as u32, lsl shift
+      );
+      address >>= 16;
+      shift += 16;
+    }
+    dynasm!(self.assembler
+        ; .arch aarch64
+        ; br x8
+    );
+  }
+
+  fn is_recv_overriden(&self) -> bool {
+    self.integer_params > 0
+  }
+
+  fn finalize(self) -> ExecutableBuffer {
+    self.assembler.finalize().unwrap()
+  }
+}
+
+struct Win64 {
+  assembler: dynasmrt::x64::Assembler,
+  params: u32,
+  offset_trampoline: u32,
+  offset_original: u32,
+  stack_allocated: u32,
+}
+
+#[cfg_attr(
+  not(all(target_aarch = "x86_64", target_family = "windows")),
+  allow(dead_code)
+)]
+impl Win64 {
+  // > Integer valued arguments in the leftmost four positions are passed in left-to-right order in RCX, RDX, R8, and R9
+  // > [...]
+  // > Any floating-point and double-precision arguments in the first four parameters are passed in XMM0 - XMM3, depending on position
+  const REGISTERS: u32 = 4;
+
+  fn new() -> Self {
+    Self {
+      assembler: dynasmrt::x64::Assembler::new().unwrap(),
+      params: 0,
+      // trampoline caller's return address + trampoline's shadow space
+      offset_trampoline: 8 + 32,
+      offset_original: 8 + 32,
+      stack_allocated: 0,
+    }
+  }
+
+  fn compile(sym: &Symbol) -> Trampoline {
+    let mut compiler = Self::new();
+
+    let can_tailcall = !compiler.must_cast_return_value(sym.result_type);
+    if !can_tailcall {
+      compiler.allocate_stack(&sym.parameter_types);
+    }
+
+    for argument in &sym.parameter_types {
+      compiler.move_left(argument)
+    }
+    if !compiler.is_recv_overriden() {
+      // the receiver object should never be expected. Avoid its unexpected or deliverated leak
```

Comment:
**nit**: "deliberate"

Reply:
installed Code Spell Checker and corrected these and some other spelling errors that you were holding back 😜 😅